wordpress-mobile/WordPress-Android
# Editor Onboarding: Enable tooltip for 50% of users

Enables the editor onboarding tooltip for 50% of users. The process to determine if the user can view the editor onboarding tooltip has also been updated to do the following:
1. Get the ID of the user. If the user logged into the app via the site address option for a self-hosted site (i.e - not a WPcom account), then we use the anonymous `userId` generated by Tracks and saved in `SharedPrefs`. 
2. Append "can_view_editor_onboarding" to the `userId` as a seed to keep the calculated hashed result unique to this feature.
3. Hash the `userId+seed` using SHA-256
4. Convert the hash byte array to a `BigInteger`
5. Divide the `BigInteger` by 100 to get the _remainder_, which will be a value between 0 and 100. 
6. If the result from step 5 is greater than or equal to 50, then the user can view the onboarding tooltip.

### To test:
The test steps in this section only verify that the onboarding tooltipis displayed properly when the editor is opened for the very first time after a fresh install to ensure nothing was broken in the process of adding the changes included in this PR. To actually verify the changes in this PR, I'd recommend logging the results from the new `getUserId()` and `convertUserIdToHash()` methods and watching the console output while working through these test steps. The `getUserId()` method should NOT return `null` or an empty or `0` string. The `convertUserIdToHash()` method should return a value between 0 and 100. 

Repeat the following test steps using a **WPcom account**, and then logging into a **self-hosted site** (non-Jetpacked) **using the site address option**. 

1. Uninstall WordPress from the test device and then install using the build from this PR or branch.
2. Log into the app.
3. Create a new blog post. Verify the **Tap to add content** tooltip appears directly above the inserter button (`+`).
4. Close the editor to return to My Site page. 
5. Tap the button to create another new blog post. Verify the tooltip does not appear. 

## Regression Notes
1. Potential unintended areas of impact: NONE.

2. What I did to test those areas of impact (or what existing automated tests I relied on): N/A

3. What automated tests I added (or what prevented me from doing so): N/A

PR submission checklist:
